web3.js/src/util/send-and-confirm-transaction.ts

```diff
@@ -27,23 +45,27 @@ export async function sendAndConfirmTransaction(
     maxRetries: options.maxRetries,
   };
 
+  let status;
+
   const signature = await connection.sendTransaction(
     transaction,
     signers,
     sendOptions,
   );
 
-  const status = (
+  status = (
     await connection.confirmTransaction(
-      signature,
+      {
+        signature: signature,
+        blockhash: transaction.recentBlockhash!,
+        lastValidBlockHeight: transaction.lastValidBlockHeight!,
+      },
```

Comment:
If anyone hands you a transaction that uses the old, deprecated call signature, this code will fatal because of a missing `lastValidBlockHeight`.

What you have to do here is to _check_ whether you have what's necessary to make a new-style call or not.

Reply:
```
const confirmationStrategy = transaction.recentBlockhash != null && transaction.lastValidBlockHeight != null
  // New, blockheight strategy.
  ? {signature, blockhash: transaction.recentBlockhash, lastValidBlockHeight: transaction.lastValidBlockHeight}
  // Old, deprecated style.
  : signature;

status = (
  await connection.confirmTransaction(confirmationStrategy, ...)
).value;
```

Reply:
Trying to work on this approach. Currently throwing an error because the overloads don't match.